plugin.program.autowidget/resources/lib/common/utils.py

```diff
@@ -392,6 +418,50 @@ def get_files_list(path, titles=None):
                 
         return new_files
 
+def cache_files(path):
+    hash = hashlib.sha1(path).hexdigest()
+    cache_path = os.path.join(_addon_path, '{}.cache'.format(hash))
+    version = _get_json_version()
+    props = version == (10, 3, 1) or (version[0] >= 11 and version[1] >= 12)
+    props_info = info_types + ['customproperties']
+    params = {'jsonrpc': '2.0', 'method': 'Files.GetDirectory',
+              'params': {'properties': info_types if not props else props_info,
+                         'directory': path},
+              'id': 1}
+    files_json = call_jsonrpc(json.dumps(params))
+    files = json.loads(files_json)
+    write_json(cache_path, files)
+    expiry = cache_expiry(hash, add=hashlib.sha1(files_json).hexdigest())
+    log("Wrote cache (exp in {}s): {}".format(expiry-time.time(), hash), 'notice')
+    return files
+
+def cache_expiry(hash, add=None):
+    # TODO: clean up no longer used cache files. not read in > 30 days?

```

Comment:
I would probably use 14 days, as that's the default cache expiry for TMDbHelper (one of the more popular add-ons to use in conjunction with AW).